curator-framework/src/main/java/org/apache/curator/framework/imps/CuratorFrameworkImpl.java

```diff
@@ -620,6 +621,15 @@ ZooKeeper getZooKeeper() throws Exception
         return client.getZooKeeper();
     }
 
+    Object getZooKeeperAdmin() throws Exception
```

Comment:
If we return client.getZooKeeper(), just like in getZooKeeper(), then this method seems unneeded to me. Clients should call getZooKeeper() and cast to ZooKeeperAdmin if they need (like in ReconfigBuilderImpl). 
You also could promise in its javadoc to return ZooKeeperAdmin for versions 3.5 and up.

Reply:
This method does the `isZk34CompatibilityMode()` check though. You don't see value in that?

Reply:
Only if the caller knows it has 3.5+ then it will be able to cast to ZooKeeperAdmin (which it currently has to anyways) and consequently use it. If it has 3.4 - it won't be able to cast - at compile time (no such class), which is safer.